lib/joystick.dart

```diff
@@ -138,25 +139,35 @@ class _JoystickState extends State<Joystick> {
     return pos.distanceTo(fpos) <= widget.pointerSize;
   }
 
+  /// Listener that is called when the controller detects motion.
+  void controllerListener(event) {
+    setState(() {
+      pointerPos = deregularize(event.data);
+      widget.onDrag(event.data);
+    });
+  }
+
   @override
   void initState() {
     super.initState();
-    // widget.controller?.init();
-    _subscription = widget.controller?.stream?.listen((event) {
-      setState(() {
-        pointerPos = deregularize(event.data);
-        widget.onDrag(event.data);
-      });
-    });
+    _subscription = widget.controller?.stream?.listen(controllerListener);
```

Comment:
Add handling for controller onDone (controller being disconnected etc)

Reply:
Controller being disconnected looks just like the user taking their finger off the screen to the virtual joystick so it should be fine.